h2/stream.py

```diff
@@ -1020,17 +1031,42 @@ def stream_reset(self, frame):
 
         return [], events
 
+    def _build_hdr_validation_flags(self, events):
+        """
+        Constructs a set of header validation flags for use when normalizing
+        and validating header blocks.
+        """
+        try:
+            is_trailer = isinstance(
+                events[0], (_TrailersSent, TrailersReceived)
+            )
+        except IndexError:
+            is_trailer = False
```

Comment:
There are cases where we don't emit an internal event &ndash; push promise is the example that springs to mind &ndash; and so `events` is empty. It's the places where we still have `assert not events`.